corehq/apps/smsbillables/models.py

```diff
@@ -353,3 +373,41 @@ def create(cls, message_log, api_response=None):
         billable.save()
 
         return billable
+
+    @classmethod
+    def _get_twilio_charges(cls, billable, message_log, backend_instance):
+        def _get_twilio_client(twilio_backend_id):
+            twilio_backend = SQLMobileBackend.get(twilio_backend_id)
+            account_sid = twilio_backend.account_sid
+            auth_token = twilio_backend.auth_token
+            return TwilioRestClient(account_sid, auth_token)
+
+        currency = None
+        try:
+            if message_log.backend_message_id:
+                twilio_message = _get_twilio_client(
+                    backend_instance
+                ).messages.get(message_log.backend_message_id)
```

Comment:
we should first check the status of the message and if it's accepted, queued, sending, or receiving, then we should stop and retry the billable task later